### DISCUSSION TITLE
Mobile Support

### DISCUSSION THREAD

**Discussion by apotenza92** [vslinko/obsidian-outliner#124] [Archive]
Hi! Is there a particular reason this plugin is now flagged as desktop only?
![image](https://user-images.githubusercontent.com/53612685/117389784-440e2200-af30-11eb-8572-df9f22648173.png)

Really keen to start using this on mobile Obsidian too!

**Comment by vslinko** [comment]
@apotenza92 Hi!

There are two reasons why Outliner isn't supporting Obsidian Mobile right now:
- Obsidian Desktop uses CodeMirror 5 but Obsidian Mobile uses CodeMirror 6. Outliner should be almost fully rewritten to support CodeMirror 6.
- UX for Obsidian Mobile should be designed from the ground because there are no keystrokes in Obsidian Mobile.

But I'll keep this idea for future.

**Reply by wanyest** [reply]
I'd love this to happen!

**Reply by wyatt** [reply]
+1

**Reply by danielbarl** [reply]
+1

**Reply by j0napkm** [reply]
+1

**Reply by KicsiMonkey** [reply]
+1

**Reply by Maleph** [reply]
+1

**Comment by lucifr** [comment]
+1 from a iPad user

**Reply by zackzeng00** [reply]
+1 from iPad user

**Comment by kdo1234** [comment]
@vslinko Hi, I'm interested in adding support for iOS, e.g. for iPad with a keyboard. Have you given thought on how support should be added for CodeMirror 6?

**Reply by vslinko** [reply]
Hello, I saw that the CodeMirror API has changed a lot and that it will be difficult to move to version 6. I haven't thought about how to do it yet, but I'm sure a bunch of hacks have to be added.

**Reply by riordan** [reply]
Gotta say, I’m blown away by @vslinko’s work on the live preview mode on desktop, which is powered by CodeMirror 6.

Perhaps someone here with the insider android build could build the plug-in and flip the desktop flag off, then report back here.

**Reply by griffimu** [reply]
> Hello, I saw that the CodeMirror API has changed a lot and that it will be difficult to move to version 6. I haven't thought about how to do it yet, but I'm sure a bunch of hacks have to be added.

Hi Viacheslav - the plugin is amazing.  Thanks for all if your work on it.  I have a question - do the recent updates you made bring mobile support closer to reality?  I'd really love to use this on my iPad.  It would make a huge positive impact on my workflow.  

Thanks,
Matt

**Comment by vslinko** [comment]
Good news for everyone! Mobile support will be released soon, after the next public release of Obsidian Mobile!

**Reply by koreyspace** [reply]
Thank you so much!

**Reply by KFrancoD** [reply]
Thanks a million!

**Reply by kngautham** [reply]
Very excited about it. Any idea on when the next public Obsidian Mobile version be released?

**Comment by alexlyzhov** [comment]
Seems like Shift+Enter for a new line within a block is unavailable on mobile (I may be wrong though), so it would be useful to introduce a command for this.

**Reply by kngautham** [reply]
They have released the new CM6 version of Obsdian Mobile for public access. Will you be releasing the mobile compatible plugin soon?

**Reply by vslinko** [reply]
I saw the recent public release of Obisidian Mobile. Unfortunately, I don't have time to update the plugin right now. I hope to be able to do so in the near future.

**Reply by kngautham** [reply]
thank you very much. will wait for your update

**Comment by ahmetasabanci** [comment]
Just saw the recent update and was excited to see if the mobile support has arrived. But looks like it was not part of this update. Hope it's going to be available soon. :)

**Comment by FritzFox** [comment]
I saw a video of the [left handed ninja](https://youtu.be/6Hn1_UMTgPo) :D .  I really enjoy working with Bulletlist.  Fast indenting and outdenting And I really like the ability to move lines around quickly.  As far as I'm concerned, the plugin is happy to integrate it into Obsidian as a standard.  Well, I admire that the people here do so much and such a good job.  Of course I hope that Outliner will also work for ipads soon :)

**Comment by Koderok** [comment]
Hi, any idea when it's going to happen? I still can't use it on the iPhone app. This and zoom plugins show up as disabled.

**Reply by ichmoimeyo** [reply]
I was able to **manually** install **zoom** on Obsidian for Android

https://github.com/vslinko/obsidian-zoom

> Manual installation
> Download main.js, manifest.json, styles.css from the [latest release](https://github.com/vslinko/obsidian-zoom/releases/latest) and put them into <vault>/.obsidian/plugins/obsidian-zoom folder.

I changed line 9
In manifest.json to ...
**"isDesktopOnly": false**

I changed line 1096
in main.js to ...
**const config = Object.assign({ legacyEditor: false }, this.app.vault.config);**

It might take a one-time additional tweak described here ...
https://github.com/vslinko/obsidian-zoom/issues/63#issuecomment-1115916085

PS: I haven't tried installing **outliner**

**Comment by jamie9090** [comment]
This is all I need to have obsidian be as strong as desktop on iPad Os 

Would love to see this!!

**Comment by kngautham** [comment]
If you install outliner beta through BART, you can move up and down the
lists in mobile too.

**Comment by jamie9090** [comment]
Oh nice, I may have to look at that. Could you point me in the right
direction for that, please?

This plugin makes Obsidian for me, love it! They need to bring it into the
standard!


Sent via Superhuman ***@***.***>

**Comment by kngautham** [comment]
Install a plugin called BRAT.
Inside BRAT Plugin, type https://github.com/vslinko/obsidian-outliner
You can use the outliner plugin after this as you normally would.

**Reply by eoiles** [reply]
"outliner doesn't support legacy editor since 1.0 please diable this option"

how to fix it?

**Reply by eoiles** [reply]
![IMG_20220709_234029.jpg](https://user-images.githubusercontent.com/24559976/178112693-19139369-1b41-40ec-8edc-715bb9ea150a.jpg)

**Comment by jamie9090** [comment]
Thanks for your help!

**Comment by pikatwinky** [comment]
Is support for Android coming soon?

**Comment by sscrewston** [comment]
I hope Outliner support for Obsidian Mobile/iPad will come soon. Really like how Logseq outliner feature works, but love Obsidian's entire workflow.

**Comment by Fyzu** [comment]
Hi all, I added support for mobile devices, and would be happy to hear feedback :)

**Reply by kankaristo** [reply]
Just installed on mobile, added commands to the mobile toolbar, and the commands appear to be working well. :smile:

This extension is really nice on mobile, because text selection and copy-pasting (just to rearrange a list item) is so cumbersome.

It would be nice to be able to "extract" the current list item (select the text and cut it to the clipboard), but that's not really related to the mobile support.

**Comment by sscrewston** [comment]
Awesome! It’s it live?

**Reply by Fyzu** [reply]
Yep!

**Comment by sscrewston** [comment]
Smooth like butter so far.

**Comment by Olanrelekan96** [comment]
Finally,  it's compatible with mobile.

**Reply by FelipeRearden** [reply]
There a PR made by the spectacular @Fyzu !!!!!!! You are a legend @Fyzu !!!!

This is the link for the PR → https://github.com/vslinko/obsidian-zoom/pull/78

All we have to do now is 🙏🙏🙏🙏🙏  :)

**Comment by Olanrelekan96** [comment]
Are you working on making zoom plugin compatible on mobile also?

**Reply by Fyzu** [reply]
Yes. Coming soon!

**Reply by Olanrelekan96** [reply]
Great

**Reply by Fyzu** [reply]
Updated!

**Reply by Olanrelekan96** [reply]
Testing it. Working great so far.

**Comment by Fyzu** [comment]
Anyone who has not yet tried the plugin on a mobile, I advise you to try it!
![ezgif-3-ac2b473541](https://user-images.githubusercontent.com/6418302/198834046-2973b729-f098-4a1e-adb1-8901f1cb85c6.gif)

**Comment by emekaelo** [comment]
I just installed and it doesn't work on my mobile.
